File: src/main/java/in/costea/wiles/commands/TokenCommand.java
package in.costea.wiles.commands;

import in.costea.wiles.data.CompilationExceptionsCollection;
import in.costea.wiles.data.Token;
import in.costea.wiles.services.TokenTransmitter;
import in.costea.wiles.statics.Constants;

import java.util.ArrayList;
import java.util.List;

public class TokenCommand extends AbstractCommand{
    private final Token token;

    public TokenCommand(TokenTransmitter transmitter, Token token)
    {
        super(transmitter);
        this.token = token;
        name = token.content();
    }

    public Token getToken()
    {
        return token;
    }

    @Override
    public Constants.SYNTAX_TYPE getType()
    {
        return Constants.SYNTAX_TYPE.TOKEN;
    }

    @Override
    public List<ExpressionCommand> getComponents()
    {
        return new ArrayList<>();
    }

    @Override
    public CompilationExceptionsCollection process()
    {
        return new CompilationExceptionsCollection();
    }

    @Override
    public String toString()
    {
        return name;
    }

}

File: src/main/java/in/costea/wiles/commands/TokenCommand.kt
package `in`.costea.wiles.commands

import `in`.costea.wiles.data.CompilationExceptionsCollection
import `in`.costea.wiles.data.Token
import `in`.costea.wiles.enums.SyntaxType
import `in`.costea.wiles.services.TokenTransmitter
import org.apache.commons.lang3.StringEscapeUtils

class TokenCommand(transmitter: TokenTransmitter, val token: Token) : AbstractCommand(transmitter) {

    init {
        name = token.content
    }

    override val type: SyntaxType
        get() = SyntaxType.TOKEN

    override fun getComponents(): List<AbstractCommand> {
        return ArrayList()
    }

    override fun process(): CompilationExceptionsCollection {
        return CompilationExceptionsCollection()
    }

    override fun toString(): String {
        return StringEscapeUtils.escapeJava(name)
    }
}